### PR TITLE
Unify deployASC and deployASA as per SPEC

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: packages/algob/src/lib/tx.ts [hunk 1]
```diff
@@ -36,11 +36,25 @@ export async function mkSuggestedParams (
 export function makeAssetCreateTxn (
   name: string, asaDef: ASADef, flags: ASADeploymentFlags, txSuggestedParams: tx.SuggestedParams
 ): tx.Transaction {
+  // Load Note
+  // Load payFlags.note (ignored if payFlags.noteb64 is present)
+  // undefined if none of them is present.
+  // If TxParams has noteb64 or note , it gets precedence
```
Comment (robert-zaremba) [inline packages/algob/src/lib/tx.ts:42]: let's keep only the last one (and remove lines 39-41)

File: packages/algob/src/lib/tx.ts [hunk 2]
```diff
@@ -36,11 +36,25 @@ export async function mkSuggestedParams (
...
   const encoder = new TextEncoder();
+  let note;
+  if (flags.noteb64 ?? flags.note) {
+    // TxParams note
+    note = flags.noteb64 ? encoder.encode(flags.noteb64) : encoder.encode(flags.note);
+  } else {
+    // ASA definition note
+    note = asaDef.noteb64 ? encoder.encode(asaDef.noteb64)
+      : (asaDef.note ? encoder.encode(asaDef.note) : undefined);
```
Comment (robert-zaremba) [inline packages/algob/src/lib/tx.ts:51]: let's create and use a function (because we have a repetition )
```
note = decodeNote(note, note64)
```

Comment (robert-zaremba) [reply packages/algob/src/lib/tx.ts:51]: Also, please add a test to that function.

**Review by robert-zaremba — APPROVED** [review]
Approving, but left 2 change requests.

File: packages/algob/src/lib/tx.ts (hunk 1 above)
Comment (amityadav0) [reply packages/algob/src/lib/tx.ts:42]: ok

File: packages/algob/src/lib/tx.ts (hunk 2 above)
Comment (amityadav0) [reply packages/algob/src/lib/tx.ts:51]: did you mean encodeNote ?

Comment (amityadav0) [reply packages/algob/src/lib/tx.ts]: Done created encodeNote function with test

**Pull request merged**